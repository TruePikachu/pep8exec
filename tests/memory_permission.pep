	LDA	0,i
	STBYTEA	0x0000,d
	LDBYTEA	0xFFFF,d
	STBYTEA	lastByte,d
	LDA	0xFFFD,d
	STA	oldWord,d
	LDA	0x8080,i
	STA	0xFFFD,d
	STA	0xFFFF,d
	LDBYTEA	0x0000,d
	ANDA	0xFF,i
	CPA	0x80,i
	BREQ	wrapped
	CPA	0x00,i
	BREQ	overRAM
	BR	noRAM
wrapped:STRO	aWrap,d
	BR	checkROM
overRAM:STRO	aOverRAM,d
	BR	checkROM
noRAM:	STRO	aNoRAM,d
checkROM:LDA	0xFFFD,d
	CPA	oldWord,d
	BREQ	noROM
	STRO	aOverROM,d
	STOP
noROM:	STRO	aNoROM,d
	STOP
lastByte:.BYTE	0
oldWord:.WORD	0
aWrap:	.ASCII	"0xFFFF write wraps\n\x00"
aOverRAM:.ASCII	"RAM overwrites\n\x00"
aNoRAM:	.ASCII	"RAM is R/O\n\x00"
aOverROM:.ASCII	"ROM overwrites\n\x00"
aNoROM:	.ASCII	"ROM is R/O\n\x00"
	.END
