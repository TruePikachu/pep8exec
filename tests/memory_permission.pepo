C0 00 00 F1 00 00 D1 FF FF F1 00 50 C1 FF FD E1
00 51 C0 80 80 E1 FF FD E1 FF FF D1 00 00 90 00
FF B0 00 80 0A 00 30 B0 00 00 0A 00 36 04 00 3C
41 00 53 04 00 3F 41 00 67 04 00 3F 41 00 77 C1
FF FD B1 00 51 0A 00 4C 41 00 83 00 41 00 93 00
00 00 00 30 78 46 46 46 46 20 77 72 69 74 65 20
77 72 61 70 73 0A 00 52 41 4D 20 6F 76 65 72 77
72 69 74 65 73 0A 00 52 41 4D 20 69 73 20 52 2F
4F 0A 00 52 4F 4D 20 6F 76 65 72 77 72 69 74 65
73 0A 00 52 4F 4D 20 69 73 20 52 2F 4F 0A 00 zz
