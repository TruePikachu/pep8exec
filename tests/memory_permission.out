RAM overwrites
ROM is R/O
